LEAF_Request_Portal/sources/Email.php

```diff
@@ -324,9 +346,22 @@ public function sendMail(): bool
             exec("php {$currDir}/../mailer/mailer.php {$emailQueueName} > /dev/null &");
         }
 
+        $this->logEmailSent($recordID);
```

Comment:
What happens here if no $recordID is being passed through. It seems you are putting a requirement on sendMail to have a recordID now. There are alot of custom scripts out there that use sendMail and don't pass through a recordID. Make sure this has backwards compatibility as well as all the other changes elsewhere in this ticket especially when you change a parameter set on a function and don't give it a default value.

Reply:
I added a default of null and allowed the parameter to accept null. The email won't be tracked if a recordID is not supplied so I guess we'll deal with HD tickets for this as they come in.